maintenance/wikia/Forum/fixCommentIndexes.php

```diff
@@ -0,0 +1,308 @@
+		// being made while this script runs
+		$this->renumberStart = $this->currentIndex + self::RACE_BUFFER;
+
+		// Set the start for new comments after this script runs to where we started renumbering plus
+		// the total number of comments we have (worst possible case of all comments being dups)
+		$this->newWallCount = $this->renumberStart + $this->commentCount;
+	}
+
+	/**
+	 * Get the index to be used for the next new comment
+	 *
+	 * @return int
+	 */
+	private function getCurrentIndex() {
+		$index = wfGetWikiaPageProp( WPP_WALL_COUNT, $this->threadId );
+		return empty( $index ) ? 0 : $index;
+	}
+
+	/**
+	 * Get the count of comments made on the thread given by $threadId.  We do this as a separate query rather
+	 * call PHP's count on the comments we get later so that we can quickly update the starting comment
+	 * index before we do anything else.
+	 *
+	 * @return array
+	 */
+	private function getCommentCount() {
+		$dbr = wfGetDB( DB_SLAVE );
+		$count = ( new WikiaSQL )
+			->SELECT( 'count(*)' )->AS_( 'count' )
+			->FROM( 'comments_index' )
+			->WHERE( 'parent_comment_id' )->EQUAL_TO( $this->threadId )
+			->ORDER_BY( 'comment_id' )
+			->runLoop( $dbr, function ( &$count, $row ) {
+				$count = $row->count;
+			});
+
+		// Ensure that we're always sending an array back
+		return empty( $count ) ? 0 : $count;
+	}
+
+	/**
+	 * Fix the comment numbering for the thread given by $this->threadId
+	 */
+	public function fix() {
+		FixCommentIndexes::debug( "* Fixing thread ".$this->threadId );
+
+		$this->setNewWallCount();
+		$this->renumberComments();
+	}
+
+	/**
+	 * Sets the new index to use whenever the next comment is made on the current thread
+	 */
+	private function setNewWallCount() {
+		if ( FixCommentIndexes::isTest() ) {
+			return;
+		}
+
+		wfSetWikiaPageProp( WPP_WALL_COUNT, $this->threadId, $this->newWallCount );
+	}
+
+	/**
+	 * Update the comment number used for creating links to those comments.  It doesn't matter what they are
+	 * as long as each comment has a unique number, however ascending from 2 is how it works now and makes the
+	 * most sense.
+	 */
+	private function renumberComments() {
+		$comments = $this->getCommentIds();
+
+		// Output some info if we're verbose.  Note that count($comments) could be higher than our
+		// $this->commentCount if anyone has commented while this script is running
+		FixCommentIndexes::debug(" -- ".$this->getThreadURL()."\n" );
+		FixCommentIndexes::debug( "-- Found ".count($comments)." comments\n" );
+		FixCommentIndexes::debug( "-- Renumbering ... " );
+
+		$commentIdx = $this->renumberStart;
+		$lastIdx = 0;
+		$foundDuplicates = false;
+
+		foreach ( $comments as $commentId ) {
+			$thisIdx = wfGetWikiaPageProp( WPP_WALL_COUNT, $commentId );
+			$isDuplicate = $thisIdx == $lastIdx;
+			$lastIdx = $thisIdx;
+
+			// If we found the start of duplicate indexes, start renumbering
+			if ( $isDuplicate ) {
```

Comment:
@kvas-damian, the links are already broken; we're just making sure they can be used in the future.